docs/excel/custom-functions-batching.md

```diff
@@ -0,0 +1,214 @@
+    args: args,
+    resolve: undefined,
+    reject: undefined,
+  };
+
+  // Create a unique promise for this invocation,
+  // and save its resolve and reject functions into the invocation entry.
+  const promise = new Promise((resolve, reject) => {
+    invocationEntry.resolve = resolve;
+    invocationEntry.reject = reject;
+  });
+
+  // Push the invocation entry into the next batch.
+  _batch.push(invocationEntry);
+
+  // If a remote request hasn't been scheduled yet,
+  // schedule it after a certain timeout, e.g. 100 ms.
+  if (!_isBatchedRequestScheduled) {
+    _isBatchedRequestScheduled = true;
+    setTimeout(_makeRemoteRequest, 100);
+  }
+
+  // Return the promise for this invocation.
+  return promise;
+}
+```
+
+## Make the remote request
+
+The purpose of the `_makeRemoteRequest` function is to pass the batch of operations to the remote service, and then return the results back to each custom function. It first creates a copy of the batch array. This allows concurrent custom function calls from Excel to immediately begin batching in a new array. The copy is then turned into a simpler array that does not contain the promise information. It wouldn't make sense to pass the promises to a remote service since they would not work. The `_makeRemoteRequest will either reject or resolve each promise based on what the remote service returns.
+
+### Add the following _makeRemoteRequest method to functions.ts
+
+```typescript
+function _makeRemoteRequest() {
+  // Copy the shared batch and allow the building of a new batch while you are waiting for a response.
+  // Note the use of "splice" rather than "slice", which will modify the original _batch array
+  // to empty it out.
+  const batchCopy = _batch.splice(0, _batch.length);
+  _isBatchedRequestScheduled = false;
+
+  // Build a simpler request batch that only contains the arguments for each invocation.
+  const requestBatch = batchCopy.map((item) => {
+    return { operation: item.operation, args: item.args };
+  });
+
+  // Make the remote request.
+  _fetchFromRemoteService(requestBatch)
+    .then((responseBatch) => {
+      // Match each value from the response batch to its corresponding invocation entry from the request batch,
+      // and resolve the invocation promise with its corresponding response value.
+      responseBatch.forEach((response, index) => {
+        if (response.error) {
+          batchCopy[index].reject(new Error(response.error));
+        } else {
+          console.log(response);
+          batchCopy[index].resolve(response.result);
+        }
+      });
+    });
+}
+```
+
+### Modify _makeRemoteRequest for your own solution
+
+The `_makeRemoteRequest` function calls `_fetchFromRemoteService` which as you'll see later is just a mock representing the remote service. This makes it easier to study and run the code in this article. But when you want to use this code for an actual remote service you should make the following changes:
```

Comment:
```suggestion
The `_makeRemoteRequest` function calls `_fetchFromRemoteService` which, as you'll see later, is just a mock representing the remote service. This makes it easier to study and run the code in this article. But when you want to use this code for an actual remote service you should make the following changes:
```
Is there a different word you can use instead of "mock?" Sample, demo, facsimile or some noun of that nature. "Mock" doesn't translate well because it has so many negative synonyms.

Reply:
@lindalu-MSFT **mock** is a standard part of unit testing. You create a mock to simulate something such as a service, database, or another system. Even though we are not creating a unit test, this is serving a very similar purpose and I thought the word would resonate better with developers. Does that work? Thanks!

Reply:
I'm just worried about localization, which will likely translate the term as a verb instead of a noun, as in to ridicule or scorn. Hopefully, they'll tag the term in keep it in English. So let's go with it if you feel good about it. Thanks for the explanation!